library/core/src/slice/mod.rs

```diff
@@ -382,16 +382,11 @@ impl<T> [T] {
     #[stable(feature = "slice_first_last_chunk", since = "1.77.0")]
     #[rustc_const_stable(feature = "slice_first_last_chunk", since = "1.77.0")]
     pub const fn split_first_chunk<const N: usize>(&self) -> Option<(&[T; N], &[T])> {
-        if self.len() < N {
-            None
-        } else {
-            // SAFETY: We manually verified the bounds of the split.
-            let (first, tail) = unsafe { self.split_at_unchecked(N) };
+        let Some((first, tail)) = self.split_at_checked(N) else { return None };
```

Comment:
The code for all of these can be made more concise by using `?`:
```suggestion
        let (first, tail) = self.split_at_checked(N)?;
```

Reply:
These functions are `const fn` so I don't believe `?` is available to use.

That results in the following error:
```rust
error[E0015]: `?` is not allowed on `Option<(&[T], &[T])>` in constant functions
   --> library\core\src\slice\mod.rs:385:29
    |
385 |         let (first, tail) = self.split_at_checked(N)?;
    |                             ^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: calls in constant functions are limited to constant functions, tuple structs and tuple variants

```

Reply:
Good point! I'd forgotten about that.